### PR TITLE
Implement Mayhem

### PR DESCRIPTION
**Pull request by nottirb**
- Added a `Dockerfile` for fuzz testing
- Added `Mayhemfile(s)` for fuzz testing
- Added a `mayhem.yml` file for automatically running fuzz tests with Mayhem using the mcode action
- Wrote 2 fuzz harnesses
    - One allows for much faster fuzzing and is decent at finding new defects
    - One allows for much slower fuzzing and is much better at analyzing all paths for a test input

### CODE REVIEW

File: fuzz/fuzz_targets/fuzz_process_deep.rs [hunk 1]
```diff
@@ -0,0 +1,240 @@
+use honggfuzz::fuzz;
```
Comment (xansec) [inline fuzz/fuzz_targets/fuzz_process_deep.rs:1]: Performance may degrade for sufficiently large input indices. I recommend putting a cap on the for loop from 0..=input.len() to something fixed.

File: fuzz/fuzz_targets/fuzz_process_rand.rs [hunk 2]
```diff
@@ -0,0 +1,257 @@
+use honggfuzz::fuzz;
```
Comment (xansec) [inline fuzz/fuzz_targets/fuzz_process_rand.rs:1]: Same comment as previous fuzz target. Also looking for confirmation that the seeded random is guaranteed to be fixed.

**Review by xansec — CHANGES_REQUESTED** [review]
The testsuites are also a bit large, I wouldn't commit more than 10-20 in general. If you feel it is necessary we can keep. Please make the requested changes.

**Comment by nottirb** [comment]
I removed the image name from the Mayhemfiles, and replaced StdRng with Pcg64 to ensure that it generates the same random number on different builds/platforms.

Also modified the deep process fuzzer to use a max of 100 split indices (randomized). Signficantly reduced the testsuite size.

Workflow might error (github packages is down as of writing this comment). I verified everything was working using dockerhub for the image with a manual mayhem run. Can re-run jobs if needed.

**Review by xansec — APPROVED** [review]
LGTM!

**Pull request merged**